src/client-ocurl/common_.ml

```diff
@@ -15,12 +15,30 @@ let debug_ =
 
 let default_url = "http://localhost:4318"
 
-let url =
-  ref (try Sys.getenv "OTEL_EXPORTER_OTLP_ENDPOINT" with _ -> default_url)
-
-let get_url () = !url
-
-let set_url s = url := s
+let make_get_from_env env_name =
+  let value = ref None in
+  fun () ->
+    match !value with
+    | None ->
+      value := Sys.getenv_opt env_name;
+      !value
+    | Some value -> Some value
```

Comment:
`env` variables are now resolved "on-demand" instead of at initialisation. This was needed for the tests, but in general I believe it is better?

They're also only resolved once and then cached. Not sure if that's necessary? 

https://github.com/imandra-ai/ocaml-opentelemetry/blob/1b538930aeca6eef2c81acec9e05512cb8f4a925/tests/ocurl/test_get_url.ml#L18-L21